packages/snackbar/index.tsx

```diff
@@ -0,0 +1,167 @@
+  onOpen?: () => void,
+  onBeforeClose?: (reason: string) => void,
+  onClose?: (reason: string) => void,
+  onAnnounce?: () => void,
+};
+
+type State = {
+  classes: Set<string>,
+};
+
+export class Snackbar extends React.Component<Props, State> {
+  foundation: IMDCSnackbarFoundation
+  constructor(props: Props) {
+    super(props);
+    const {timeoutMs, closeOnEscape, leading, stacked} = this.props;
+    const classes = new Set();
+    if (leading) {
+      classes.add('mdc-snackbar--leading');
+    }
+
+    if (stacked) {
+      classes.add('mdc-snackbar--stacked');
+    }
+
+    this.state = {
+      classes,
+    };
+
+    this.foundation = new MDCSnackbarFoundation(this.adapter);
+    if (timeoutMs) {
+      this.foundation.setTimeoutMs(timeoutMs);
+    }
+
+    if (closeOnEscape) {
+      this.foundation.setCloseOnEscape(closeOnEscape);
+    }
+  }
+  get adapter(): IMDCSnackbarAdapter {
+    return {
+      addClass: (className: string) => {
+        const {classes} = this.state;
+        classes.add(className);
+        this.setState({
+          classes,
+        });
+      },
+      removeClass: (className: string) => {
+        const {classes} = this.state;
+        classes.delete(className);
+        this.setState({
+          classes,
+        });
+      },
+      announce: () => {
+        // Usually it works automatically if this component uses conditional rendering
+        this.props.onAnnounce && this.props.onAnnounce();
+      },
+      notifyOpening: () => {
+        const {onBeforeOpen} = this.props;
+        if (onBeforeOpen) {
+          onBeforeOpen();
+        }
+      },
+      notifyOpened: () => {
+        const {onOpen} = this.props;
+        if (onOpen) {
+          onOpen();
+        }
+      },
+      notifyClosing: (reason: string) => {
+        const {onBeforeClose} = this.props;
+        if (onBeforeClose) {
+          onBeforeClose(reason);
+        }
+      },
+      notifyClosed: (reason: string) => {
+        const {onClose} = this.props;
+        if (onClose) {
+          onClose(reason);
+        }
+      },
+    };
+  }
+  close(action: string) {
+    this.foundation.close(action);
+  }
+  getTimeoutMs() {
+    return this.foundation.getTimeoutMs();
+  }
+  getCloseOnEscape() {
+    return this.foundation.getTimeoutMs();
+  }
+  isOpen() {
+    return this.foundation.isOpen();
+  }
+  handleKeyDown = (e: React.KeyboardEvent) => {
+    this.foundation.handleKeyDown(e.nativeEvent);
+  }
+  handleActionClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    this.foundation.handleActionButtonClick(e.nativeEvent);
+  }
+  componentDidMount() {
+    this.foundation.init();
+    this.foundation.open();
+  }
+  componentWillUnmount() {
+    this.foundation.destroy();
+  }
+  get classes() {
+    return classnames(this.props.className, ...Array.from(this.state.classes));
+  }
+  render() {
+    return <div className={`mdc-snackbar ${this.classes}`} onKeyDown={this.handleKeyDown}>
```

Comment:
Can you add `mdc-snackbar` to `this.classes()`? Then this line is just `className={this.classes}`

Reply:
But you could have 'mdc-snackbar' as the first argument passed to classnames in the classes getter, then it would always be first, would it not?

Reply:
You are right